File: 백준/Silver/10828. 스택/스택.py
import sys
from collections import deque
input=sys.stdin.readline

stack = []

N = int(input()) # 명령의 수
for _ in range(N):
    command = list(input().split())
    
    if(command[0] == 'push'):
        stack.append(int(command[1]))
    elif(command[0] == 'pop'):
        if(len(stack)==0):
            print(-1)
        else:
            print(stack.pop())
    elif(command[0] == 'size'):
        print(len(stack))
    elif(command[0] == 'empty'):
        if(len(stack)==0):
            print(1)
        else:
            print(0)
    elif(command[0] == 'top'):
        if(len(stack)==0):
            print(-1)
        else:
            print(stack[-1])
        

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>

using namespace std;

int arr[10001];
int cur = 0;

int main()
{
	int N; cin >> N;
	string op;
	int num;

	for (int i = 0; i < N; i++)
	{
		cin >> op;
		if (op == "push")
		{
			cin >> num;
			arr[cur] = num;
			cur++;
		}
		else if (op == "pop")
		{
			if (cur == 0) // 비어있음
				cout << -1 << "\n";
			else
			{
				cur--;
				cout << arr[cur] << "\n";
			}
		}
		else if (op == "size")
		{
			cout << cur << "\n";
		}
		else if (op == "empty")
		{
			if (cur == 0)
				cout << 1 << "\n";
			else
				cout << 0 << "\n";
		}
		else if (op == "top")
		{
			if (cur == 0) // 비어있음
				cout << -1 << "\n";
			else
			{
				cur--;
				cout << arr[cur] << "\n";
				cur++;
			}
		}
	}
}